src/Build/Instance/ProjectItemInstance.cs

```diff
@@ -1368,6 +1368,11 @@ public void CopyMetadataTo(ITaskItem destinationItem, bool addOriginalItemSpec)
                         {
                             // Utilities.TaskItem's don't know about item definition metadata. So merge that into the values.
                             destinationItem.SetMetadata(metadatum.Name, GetMetadataEscaped(metadatum.Name));
+                            if (metadatum.Name == "OriginalItemSpec")
```

Comment:
If the intent of `OriginalItemSpec` is to contain the value of the latest item metadata was copied from, then this would break that behaviour, as it would not start propagating the previous value.

Reply:
:tired_face:

The logic is duplicated in:
https://github.com/Microsoft/msbuild/blob/master/src/Utilities/TaskItem.cs#L367
https://github.com/Microsoft/msbuild/blob/master/src/Shared/TaskParameter.cs#L682

Searching for `SetMetadata("OriginalItemSpec"` finds other clones all over the place that patch this metadata.